internal/signalfx-agent/pkg/monitors/prometheusexporter/conversion.go

```diff
@@ -63,13 +63,8 @@ func makeSummaryDatapoints(name string, ms []*dto.Metric) []*datapoint.Datapoint
 			continue
 		}
 
-		if s.SampleCount != nil {
```

Comment:
Good catch. I was a bit unsure of what to do about this file. 

Another example of this same issue, the underlying logic changed in that there will never be a nil data type in the `convertMetricFamily` function. I removed the `mf.Type == nil` check and now we do a switch on `mf.GetType()`. However, `COUNTER` will be returned as the default data type if it's nil, which results in the same situation you're pointing out. I didn't get a chance to investigate to see if it's a valid use-case to get a nil metric type, but it **seems** like it shouldn't be valid.

Do we simply skip lint so we can keep everything the same, or try to investigate to see if we can update to match linting rules and the [called library's](https://github.com/prometheus/client_model) functionality? (Quick skimming the library's README shows it's not the most supported itself, so maybe it's not a problem to skip linting here)

Reply:
The [linter's](https://github.com/ghostiam/protogetter) objective is to avoid nil pointer dereferences when working on protobuf types, which we aren't at high risk for considering the triggering statements are nil checks themselves. imo linter changes should be transparent (only defensive or idiomatic noops) so when we "know what we're doing" or the adoption of a linter is unnecessary it's fine to disable on a case basis. User telemetry changes should not occur at the expense of adopting a linter check.

Reply:
That's fair. I'll skip lint for this file and remove changes.